pkg/river/vm/vm.go

```diff
@@ -173,26 +173,59 @@ func (vm *Evaluator) evaluateBlockOrBody(scope *Scope, assoc map[value.Value]ast
 			continue
 
 		case tf.IsAttr() && len(blocks) > 0:
-			return fmt.Errorf("%q must be an attribute, but is used as a block", fullName)
+			return diag.Diagnostic{
+				Severity: diag.SeverityLevelError,
+				StartPos: ast.StartPos(blocks[0]).Position(),
+				EndPos:   ast.EndPos(blocks[len(blocks)-1]).Position(),
+				Message:  fmt.Sprintf("%q must be an attribute, but is used as a block", fullName),
+			}
 		case tf.IsAttr() && len(attrs) == 0 && !tf.IsOptional():
-			return fmt.Errorf("missing required attribute %q", fullName)
+			return diag.Diagnostic{
+				Severity: diag.SeverityLevelError,
+				//TODO: We need to report the start and end position of the block where this attribute should be?
+				StartPos: ast.StartPos().Position(),
+				EndPos:   ast.EndPos().Position(),
```

Comment:
Use the starting and ending position of the `node` argument, which will be the entire surrounding block. 

For example, this would mean that 

```river 
prometheus.scrape {
  targets = [ ... ] 
}
```

will highlight the whole component as invalid because it is missing the "forward_to" required attribute.